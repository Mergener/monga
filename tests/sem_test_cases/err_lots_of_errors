type T = {
    f : float;
};

function foo() {
    var i : int;
    var c : R;

    i = 4.5;

    c.x = i;

    var t : T;
    t.f = 2.0;
    t.f = i;

    return 3;
}