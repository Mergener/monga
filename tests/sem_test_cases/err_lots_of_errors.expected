[Error] Module tests/sem_test_cases/err_lots_of_errors, line 7, column 15: Unknown type 'R' for variable 'c'.
[Error] Module tests/sem_test_cases/err_lots_of_errors, line 9, column 13: Cannot assign value of type 'double' to lvalue of type 'int'.
[Error] Module tests/sem_test_cases/err_lots_of_errors, line 15, column 13: Cannot assign value of type 'int' to lvalue of type 'float'.
[Error] Module tests/sem_test_cases/err_lots_of_errors, line 17, column 14: Cannot return value of type 'int' in function declared with return type 'void'.
Semantic errors were caught during analysis.
